torchtune/modules/moe/moe.py

```diff
@@ -133,6 +136,34 @@ def forward(self, x: torch.Tensor) -> torch.Tensor:
         )
         routed_input = routed_input * top_scores.reshape(-1, 1)
 
+        if self.use_grouped_mm:
+            # NOTE: In order to use torch._grouped_mm, we need to make sure
+            # the number of tokens each expert gets is a multiple of 16.
+            # The following kernel helps achieve this via padding, without
+            # incurring synchronization between device and host.
+            from torchtune.modules.moe.indices import generate_permute_indices
+
+            ALIGN_SIZE_M = 16  # noqa
+
+            with torch.no_grad():
+                (
+                    permuted_indices,
+                    num_tokens_per_expert,
+                    _,
+                ) = generate_permute_indices(
+                    num_tokens_per_expert,
+                    self.experts.num_experts,
+                    1,
```

Comment:
This is just because we are fully replicating the experts over all ranks, right?